### PR TITLE
Fixes so works out of the box with OCP 4.

### PR DESCRIPTION
**Pull request by GrahamDumpleton**
The original master did not work out of the box on a fresh OCP 4 cluster with Subscription and CatalogSource resources breaking ability to install the pipelines operator through the web UI, yet they didn't themselves install the operator. They have been removed, and instead rely on manual installation of pipelines operator into cluster. For now this means that this workshop can only be used in a shared cluster when cluster admin sets up the operator once for the cluster.

Also fixed `.workshop/scripts`, which was wrongly made a copy of files rather than required Git submodule so can easily get updates.

Various other changes made to avoid users needing to manually change commands run, or edit config files. Only remaining thing where manual editing is required is tailing `pipelinerun` log file. Deducing the name from `oc get` output is hard because not easy to work it which is running instance from resource status.

Note that even with changes, the Java application doesn't appear to work anyway. The application doesn't actually listen for HTTP requests on ports 8080/8443. Thus readiness probe fails and pod gets restarted, eventually failing.

### CODE REVIEW

**Comment by GrahamDumpleton** [comment]
Note that if you have a checkout already, because what was a physical directory has been replaced with the Git submodule, to be safe you may want to delete you checkout and clone again. In doing this, you must follow the instructions in the `README` file to ensure that the Git submodule is also checked out.

**Comment by GrahamDumpleton** [comment]
BTW, the repo should really be renamed to `lab-tekton-pipelines`. The original name is too long and generated host names under Homeroom exceed maximum for DNS host names. Things are being compensated for in `.workshop/settings.sh` file by overriding name to be shorter value of `lab-tekton-pipelines`, but would be good to rename repo to match.

**Comment by GrahamDumpleton** [comment]
Working out pipelinerun/pod name for tailing logs is now automatically handled.

As to the fact that Java application doesn't work, I suspect it is because using generic petclinic spring application. Likely the layout of the repo will mean it will not work with Java S2I on OpenShift. Should not be using that application. We should use our own application that we have control over, such as:

https://github.com/openshift-roadshow/parksmap-web

I swapped in parksmap into the YAML file as a test and it works.

Doesn't matter if not hooked up to a back end, as long as it shows map tiles is enough.

As with any Java application though, takes a long to start up so can verify it is working.

File: workshop/content/exercises/trigger-pipeline.adoc [hunk 1]
```diff
@@ -29,7 +29,7 @@ spec:
   type: image
   params:
   - name: url
-    value: image-registry.openshift-image-registry.svc:5000/pipelines-tutorial/spring-petclinic
+    value: %image_registry%/pipelines-tutorial/spring-petclinic
```
Comment (danielhelfand) [inline workshop/content/exercises/trigger-pipeline.adoc]: Can we include the namespace dynamically here as well (i.e., replace `pipelines-tutorial` with the namespace created for the user)?

**Comment by danielhelfand** [comment]
> BTW, the repo should really be renamed to `lab-tekton-pipelines`. The original name is too long and generated host names under Homeroom exceed maximum for DNS host names. Things are being compensated for in `.workshop/settings.sh` file by overriding name to be shorter value of `lab-tekton-pipelines`, but would be good to rename repo to match.

Sure, I can change it once we merge everything here in.

**Comment by danielhelfand** [comment]
I can also confirm the liveness/readiness probes fail with the following:

```
Liveness probe failed: Get http://10.128.2.66:8080/: dial tcp 10.128.2.66:8080: connect: connection refused
```

```
Readiness probe failed: Get http://10.128.2.66:8080/: dial tcp 10.128.2.66:8080: connect: connection refused
```

The sample app however is starting up appropriately in the container, which I saw by looking at the logs.

File: workshop/content/verify-workshop-environment.adoc [hunk 2]
```diff
@@ -9,9 +9,9 @@ The OpenShift Pipelines operator provides all its resources under a single API g
 oc api-resources --api-group=tekton.dev
 ----
 
-To validate that your user has been granted the appropriate roles, you can use the `oc auth can-i` command to see whether you can create Kubernetes custom resources of the kind the OpenShift Pipelines operator responds to.
+Did you type the command in yourself? If you did, click on the command instead and you will find that it is executed for you. You can click on any command which has the <span class="fas fa-play-circle"></span> icon shown to the right of it, and it will be copied to the interactive terminal and run.
```
Comment (danielhelfand) [inline workshop/content/verify-workshop-environment.adoc]: `<span class="fas fa-play-circle"></span>` isn't rendering properly:

image

**Comment by GrahamDumpleton** [comment]
The petclinic app may start, but if you get inside the container before it dies, it doesn't even respond on http://localhost:8080. So it doesn't appear to be accepting requests.

**Comment by GrahamDumpleton** [comment]
I need to fix the embedded HTML. Realised it didn't work somewhere else used it and took it out from there, but forgot had it anywhere else. Is just a matter of using AsciiDoc method for embedded HTML.

**Comment by danielhelfand** [comment]
> The petclinic app may start, but if you get inside the container before it dies, it doesn't even respond on http://localhost:8080. So it doesn't appear to be accepting requests.

I removed the liveness/readiness probes, and the app started up at the url. I am pretty sure it just takes a long time to start so the liveness and readiness probes fail because the app isn't ready to accept requests.

**Comment by GrahamDumpleton** [comment]
Which is part of the problem with relying on Java apps in demos. They take too long to build and then too long to start up, meaning even if have long timeout on probes, people think they have failed as they keep saying service unavailable.

**Comment by danielhelfand** [comment]
Right, it might be worthwhile to look into a different, non-Java sample app. We went with this since it was what was in the original tutorial. 

For now though, I think these changes look good if you're comfortable with them.

**Pull request merged**